docs/website/mkdocs.yml

```diff
@@ -196,9 +196,11 @@ nav:
           - "developers/performance/profiling-gpu-vulkan.md"
           - "developers/performance/profiling-with-tracy.md"
       - "Design docs":
+          - "developers/design-docs/cuda-hal-driver.md"
           - "developers/design-docs/design-roadmap.md"
           - "developers/design-docs/function-abi.md"
           - "developers/design-docs/invocation-execution-model.md"
+          - "developers/design-docs/metal-hal-driver.md"
```

Comment:
nit: could break the alphabetical ordering to put the two HAL driver docs next to each other